src/node_file.cc

```diff
@@ -2480,6 +2480,64 @@ static void WriteString(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+static void WriteFileUtf8(const FunctionCallbackInfo<Value>& args) {
+  // Fast C++ path for fs.writeFileSync(path, data) with utf8 encoding
+  // (file, data, options.flag, options.mode)
+
+  Environment* env = Environment::GetCurrent(args);
+  auto isolate = env->isolate();
+
+  CHECK_GE(args.Length(), 4);
+
+  CHECK(args[1]->IsString());
+
+  CHECK(args[2]->IsInt32());
+  const int flags = args[2].As<Int32>()->Value();
+
+  CHECK(args[3]->IsInt32());
+  const int mode = args[3].As<Int32>()->Value();
+
+  uv_file file;
+  uv_fs_t req;
+
+  bool is_fd = args[0]->IsInt32();
+
+  // Check for file descriptor
+  if (is_fd) {
+    file = args[0].As<Int32>()->Value();
+  } else {
+    BufferValue path(env->isolate(), args[0]);
+    CHECK_NOT_NULL(*path);
+    if (CheckOpenPermissions(env, path, flags).IsNothing()) return;
+
+    FS_SYNC_TRACE_BEGIN(open);
+    file = uv_fs_open(nullptr, &req, *path, flags, mode, nullptr);
+    FS_SYNC_TRACE_END(open);
+
+    if (req.result < 0) {
+      uv_fs_req_cleanup(&req);
+      return env->ThrowUVException(req.result, "open", nullptr, path.out());
+    }
+  }
+
+  node::Utf8Value value(isolate, args[1]);
+  uv_buf_t uvbuf = uv_buf_init(*value, value.length());
+
+  FS_SYNC_TRACE_BEGIN(write);
+  int err = uv_fs_write(nullptr, &req, file, &uvbuf, 1, 0, nullptr);
```

Comment:
Critical: doesn't (but should) handle partial writes like the JS implementation does.

Reply:
Yeah this is being discussed in this conversation: https://github.com/nodejs/node/pull/49884#discussion_r1337836307

Reply:
Huh, maybe a github glitch. There wasn't any (visible) discussion 5 minutes ago. I'll comment in the other thread.